### PR TITLE
Adding `whylogs` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
## Reason for Build Failure: `1/N`

`Observation`:

> :point_right: *Need `cmake` for building `whylogs-sketching`.*

- `line number: 992`: [(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=549733&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=992)
- `time elapsed`: `3m:18s`

```sh
943    FileNotFoundError: [Errno 2] No such file or directory: 'cmake'
...
...
...
992    RuntimeError: CMake >= 3.12 must be installed to build the following extensions: whylogs-sketching
```


<details>
<summary><b>Click to expand</b>: Detailed Error Trace</summary>
<p>

```sh

  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/conda/staged-recipes/build_artifacts/whylogs-sketching_1660243448164/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/python /home/conda/staged-recipes/build_artifacts/whylogs-sketching_1660243448164/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /tmp/tmpbckm7ojs
  cwd: /home/conda/staged-recipes/build_artifacts/whylogs-sketching_1660243448164/work
  Building wheel for whylogs-sketching (pyproject.toml): finished with status 'error'
  ERROR: Failed building wheel for whylogs-sketching
Failed to build whylogs-sketching
ERROR: Could not build wheels for whylogs-sketching, which is required to install pyproject.toml-based projects
Exception information:
Traceback (most recent call last):
  File "$PREFIX/lib/python3.10/site-packages/pip/_internal/cli/base_command.py", line 167, in exc_logging_wrapper
    status = run_func(*args)
  File "$PREFIX/lib/python3.10/site-packages/pip/_internal/cli/req_command.py", line 247, in wrapper
    return func(self, options, args)
  File "$PREFIX/lib/python3.10/site-packages/pip/_internal/commands/install.py", line 431, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: Could not build wheels for whylogs-sketching, which is required to install pyproject.toml-based projects
Removed build tracker: '/tmp/pip-build-tracker-cn07aim5'
Traceback (most recent call last):
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 223, in <module>
    build_all(os.path.join(root_dir, "recipes"), args.arch)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 120, in build_all
    build_folders(recipes_dir, folders, arch, channel_urls)
  File "/home/conda/staged-recipes-copy/.ci_support/build_all.py", line 179, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 3088, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 2211, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/utils.py", line 411, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/conda_build/utils.py", line 391, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/home/conda/staged-recipes/build_artifacts/whylogs-sketching_1660243448164/work/conda_build.sh']' returned non-zero exit status 1.
##[error]Bash exited with code '1'.
```

</p>
</details>

**Comment by sugatoray** [comment]
@BastianZim I need to include `cmake` in the dependencies. I have not done much of packaging for libraries that use `c/c++` within. Could you please help me on this one?

Thank you.

**Comment by BastianZim** [comment]
Sure, I’ll add some suggestions tomorrow. Are you happy with me pushing to your branch?

**Comment by sugatoray** [comment]
> Sure, I’ll add some suggestions tomorrow. Are you happy with me pushing to your branch?

Absolutely! I will watch and learn. Please do. :+1:

**Comment by sugatoray** [comment]
@BastianZim A gentle ping.

File: recipes/whylogs-sketching/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{% set name = "whylogs-sketching" %}
+{% set version = "3.4.1.dev2" %}
```
Comment (BastianZim) [inline recipes/whylogs-sketching/meta.yaml:2]: This needs to target the dev channel: https://conda-forge.org/docs/maintainer/knowledge_base.html#pre-release-builds

File: recipes/whylogs-sketching/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - cmake >=3.12
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python >=3.7
+    - pybind11-global >=2.6.0
```
Comment (BastianZim) [inline recipes/whylogs-sketching/meta.yaml:24]: This is all copied from the pyproject.toml file in the source distribution.

File: recipes/whylogs-sketching/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  sha256: 0b0dc9c43e0d6a87924a45a95b922402c44600527bd22affb7eeab4a4a697936
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - cmake >=3.12
+    - {{ compiler('cxx') }}
```
Comment (BastianZim) [inline recipes/whylogs-sketching/meta.yaml:20]: More info here: https://conda-forge.org/docs/maintainer/adding_pkgs.html#requirements

**Comment by BastianZim** [comment]
Sorry, completely forgot this. I basically just copied the info from the source distribution as the GitHub repo is private. I haven't run this yet but this should be a good base.

File: recipes/whylogs-sketching/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    - make  # [not win]
+    - ninja  # [win]
```
Comment (BastianZim) [inline recipes/whylogs-sketching/meta.yaml:22]: Ref: https://conda-forge.org/docs/maintainer/knowledge_base.html#using-cmake

File: recipes/whylogs/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
```
Comment (BastianZim) [inline recipes/whylogs/meta.yaml:21]:
```suggestion
    - python >=3.7
    - poetry-core>=1.0.0
```
Ref: https://github.com/whylabs/whylogs/blob/c1e88184a2e96533659f32f5285ab79aa5ec76ea/python/pyproject.toml#L108

File: recipes/whylogs-sketching/meta.yaml (hunk 1 above)
Comment (sugatoray) [reply recipes/whylogs-sketching/meta.yaml:2]: Haven't done this yet. Will do soon.

**Comment by BastianZim** [comment]
Unfortunately, Windows is something that eludes me but the rest should get you going. Maybe ask in the Gitter chat?

**Pull request closed without merging**